forest/cli/src/cli/info_cmd.rs

```diff
@@ -0,0 +1,406 @@
+        };
+
+        let wallet_balance = match balance(node_status_info.default_wallet_address_balance) {
+            Ok(bal) => bal.bold(),
+            Err(err) => err.to_string().bold(),
+        };
+
+        NodeInfoOutput {
+            chain_status,
+            network,
+            uptime,
+            health,
+            wallet_address,
+            wallet_balance,
+        }
+    }
+}
+
+impl NodeInfoOutput {
+    fn set_color(mut self, color: &LoggingColor) -> Self {
+        if !color.coloring_enabled() {
+            self.chain_status = self.chain_status.clear();
+            self.network = self.network.clear();
+            self.uptime = self.uptime.clear();
+            self.health = self.health.clear();
+            self.wallet_address = self.wallet_address.clear();
+            self.wallet_balance = self.wallet_balance.clear();
+        }
+
+        self
+    }
+}
+
+impl InfoCommand {
+    pub async fn run(&self, config: Config, opts: &CliOpts) -> anyhow::Result<()> {
+        let node_status = node_status(&config.client.rpc_token)
+            .await
+            .map_err(handle_rpc_err)?;
+
+        let head = chain_head(&config.client.rpc_token)
+            .await
+            .map_err(handle_rpc_err)?;
+
+        let network = chain_get_name((), &config.client.rpc_token)
+            .await
+            .map_err(handle_rpc_err)?;
+
+        let start_time = start_time(&config.client.rpc_token)
+            .await
+            .map_err(handle_rpc_err)?;
+
+        let cur_duration: Duration = SystemTime::now().duration_since(UNIX_EPOCH)?;
+        let blocks_per_tipset_last_finality =
+            node_status.chain_status.blocks_per_tipset_last_finality;
+
+        let mut node_status_info =
+            NodeStatusInfo::new(cur_duration, blocks_per_tipset_last_finality, head)?;
+
+        let default_wallet_address = wallet_default_address((), &config.client.rpc_token)
+            .await
+            .map_err(handle_rpc_err)?;
+
+        let default_wallet_address_balance = if let Some(def_addr) = &default_wallet_address {
+            let balance = wallet_balance((def_addr.clone(),), &config.client.rpc_token)
+                .await
+                .map_err(handle_rpc_err)?;
+            Some(balance)
+        } else {
+            None
+        };
+
+        node_status_info.start_time = start_time;
+        node_status_info.network = network;
+        node_status_info.default_wallet_address = default_wallet_address.clone();
+        node_status_info.default_wallet_address_balance = default_wallet_address_balance;
+
+        let info_output = NodeInfoOutput::from(node_status_info).set_color(&opts.color);
+
+        println!("Network: {}", info_output.network);
+        println!("Uptime: {}", info_output.uptime);
+        println!("Chain: {}", info_output.chain_status);
+        println!("Chain health: {}", info_output.health);
+        println!(
+            "Default wallet address: {} [balance: {}]",
```

Comment:
This will results in an error if we don't have a default address.

Reply:
You could print instead: `"Default wallet address: {}",`

And handle all the output formatting logic inside `NodeInfoOutput::from`.

And you could print `[balance: xxx]` string only if there is a default address set.

Reply:
Good catch, thx. Fixed.